app/search-music.js

```diff
@@ -0,0 +1,43 @@
+function SearchMusic () {
+
+}
+
+
+
+
+
+SearchMusic.prototype.SearchByValue = function(values,playlist) {
+  playlist.forEach(function(entry) {
+       entry.songs.forEach(function(song) {
+       if (values == song.getTitle()) {
+         clonare(entry.getId());
+       }
+     });
+   });
+
+/* Trying to use map and reduce
+var array_songs = (((playlist.map(getSongList)).reduce(reduceSongList,[])).
```

Comment:
not really understanding what you're doing here. trying puttin the result of each map, filter, reduce into really well named variables in order to figure out the right way of doing it.